### PR TITLE
feat(HtmlWebpackPlugin): added the option to add favicon through config/command

### LINKED ISSUE #168: Favicon support
We want to be able to customize the favicon. Whether we have a path or an actual icon, we are unable to tell the engine to use it.

Regarding Electron, I am not sure how it works but the idea is the same, we want to be able to customize the app's icon.

### PR DESCRIPTION
**Pull request by eifr0980**
# Description 
Added new option to add a favicon to `HtmlWebpackPlugin` using 2 ways:

### Command
Add `--favicon` flag and path to start/build command:
```console
foo@bar:~$ yarn start --favicon ./assets/favicon.ico
```

### Config
Add `favicon` and path to `engine.config.js`:
```js
const path = require('path');
module.exports = {
    favicon: path.join(__dirname, 'favicon.ico'),
};
```

Closes #168

### CODE REVIEW

File: packages/engineer/src/cli-commands.ts [hunk 1]
```diff
@@ -53,6 +53,7 @@ export const startCommand: Command = (program) =>
             true
         )
         .option('--title <title>', 'application title to display in browser')
+        .option('--favicon <favicon>', 'application favicon to display in browser')
```
Comment (RomanYarik) [inline packages/engineer/src/cli-commands.ts:56]: I think `faviconPath` will be better here
also about the description, maybe `path to favicon to be displayed in browser environments`

File: packages/engineer/src/cli-commands.ts [hunk 2]
```diff
@@ -83,6 +85,10 @@ export const startCommand: Command = (program) =>
             } = cmd;
 
             try {
+                const basePath = resolve(path);
+                preRequire(pathsToRequire, basePath);
```
Comment (RomanYarik) [inline packages/engineer/src/cli-commands.ts:89]: this already happens inside `startDevServer`. why did you call it here?

File: packages/engineer/src/feature/gui.dev-server.env.ts [hunk 3]
```diff
@@ -14,7 +14,7 @@ guiFeature.setup(
         {
             buildFeature: {
                 engineerWebpackConfigs,
-                devServerConfig: { title, publicConfigsRoute },
+                devServerConfig: { title, favicon, publicConfigsRoute },
```
Comment (RomanYarik) [inline packages/engineer/src/feature/gui.dev-server.env.ts:17]: why do we need it here?
this is the configuration for the engineer dashboard

**Review by RomanYarik — CHANGES_REQUESTED** [review]

File: packages/engineer/src/feature/gui.dev-server.env.ts (hunk 3 above)
Comment (eifr0980) [reply packages/engineer/src/feature/gui.dev-server.env.ts:17]: added it next to places that has `title`.
So what's the usage of `title` overthere?

Comment (RomanYarik) [reply packages/engineer/src/feature/gui.dev-server.env.ts:17]: well.. not sure, dont see any good reason.

it's fine by me to keep it like this

**Pull request merged**